backend/hitas/views/reports.py

```diff
@@ -124,3 +126,40 @@ def list(self, request: Request, *args, **kwargs) -> HttpResponse:
         workbook = build_multiple_ownerships_report_excel(ownerships)
         filename = "Useamman sääntelyn piirissä olevan asunnon omistavat omistajat.xlsx"
         return get_excel_response(filename=filename, excel=workbook)
+
+
+class OwnershipsByCompanyJSONReportView(ViewSet):
+    NON_DISCLOSURE_NAME = "xxx"
+
+    def retrieve(self, request: Request, *args, **kwargs) -> Response:
+        data = [
+            {
+                "number": d.apartment_number,
+                "surface_area": d.apartment_surface_area,
+                "share_numbers": f"{d.apartment_share_number_start}-{d.apartment_share_number_end}",
+                "purchase_date": d.sale_purchase_date,
+                "owner_name": self._get_owner_name(d.owner),
+                "owner_ssn": self._get_ssn(d.owner),
+            }
+            for d in find_apartments_by_housing_company(kwargs.get("pk"))
+        ]
+
+        return Response(data=data, status=status.HTTP_200_OK)
```

Comment:
This works, but a serializer might have been more in style with DRF 😉

Reply:
True, it was done like this in some other view so took the "style" from there :D. looks like serializers are used in other so I will change this.